central/processbaseline/service/service_impl.go

```
@@ -72,14 +77,41 @@ func (s *serviceImpl) GetProcessBaseline(ctx context.Context, request *v1.GetPro
 	}
 	baseline, exists, err := s.dataStore.GetProcessBaseline(ctx, request.GetKey())
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(errox.NotFound, "No process baseline with key %+v found", request.GetKey())
 	}
 	if !exists {
 		return nil, errors.Wrapf(errox.NotFound, "No process baseline with key %+v found", request.GetKey())
 	}
 	return baseline, nil
 }
 
+func (s *serviceImpl) GetLoadProcessBaseline(ctx context.Context, request *v1.GetProcessBaselineRequest) (*storage.ProcessBaseline, error) {
+	if err := validateKeyNotEmpty(request.GetKey()); err != nil {
+		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	}
+	baseline, exists, err := s.dataStore.GetProcessBaseline(ctx, request.GetKey())
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		// Make sure the deployment still exists before we try to build a baseline.
+		_, deploymentExists, err := s.deployments.GetDeployment(ctx, request.GetKey().GetDeploymentId())
+		if err != nil {
+			return nil, err
+		}
+		if !deploymentExists {
+			return nil, errors.Wrapf(errox.NotFound, "deployment with id '%s' does not exist", request.GetKey().GetDeploymentId())
+		}
+
+		// Build an unlocked baseline
+		baseline, err = s.dataStore.CreateUnlockedProcessBaseline(ctx, request.GetKey())
+		if err != nil || baseline == nil {
```

This ends up eating the error. let's check these separately